main/mafia.hs

```diff
@@ -304,15 +311,41 @@ pConstraint =
 
 pScriptPath :: Parser File
 pScriptPath =
-  argument textRead $
-       metavar "SCRIPT"
-    <> help "The path to the Haskell shell script."
+  argument scriptRead $
+       metavar "SCRIPT_PATH"
+    <> help "The path to a Haskell script to execute."
 
 pScriptArgs :: Parser File
 pScriptArgs =
   argument textRead $
        metavar "SCRIPT_ARGUMENTS"
-    <> help "Argument to pass to the shell script."
+    <> help "Arguments to pass to the script."
+
+scriptRead :: ReadM File
+scriptRead =
+  eitherReader $ \path ->
+    if isScriptPath path then
+      Right $ T.pack path
```

Comment:
Hmm, it's true, although I'm hoping it will be become the real code path in optparse 0.14.

I think you're right though, having dead code isn't a good idea. Curious, how does `readerError` differ from the `Left` case?